wrong-import-order for third party libraries depends on working directory

### Bug description

The working directory from where `pylint` binary was run influences whether it outputs the `wrong-import-order` coding convention error or not

### Configuration

Given the following directory structure
```ini
root_dir
└── repo_dir
    ├── pyproject.toml
    └── src
        ├── module1
        │   ├── function1.py
        │   └── __init__.py
        └── module2
            ├── function2.py
            └── __init__.py
```
with the following file contents

- pyproject.toml
```
[build-system]
requires = ["hatchling"]
build-backend = "hatchling.build"

[project]
name = "pylint-wrong-import-order-bug-minimal-example"
version = "0.0.1"
requires-python = ">=3.11"
dependencies = [
  "pylint"
]
```

- src/module1/function1.py
```
'''Module 1
'''

import module2

import pylint

def function1():
    '''Function 1
    '''
    module2.function2()
    print(f"{pylint.__version__}")

if __name__ == "__main__":
    function1()
```

- src/module1/\_\_init\_\_.py
```
'''Init for module1
'''
```

- src/module2/function2.py
```
'''Module 2
'''

def function2():
    '''Function 2
    '''
```

- src/module2/\_\_init\_\_.py
```
'''Init for module2
'''

from .function2 import function2
```

The following is only thrown when the working directory where `pylint` is run is "repo_dir"
```
************* Module module1.function1
src/module1/function1.py:6:0: C0411: third party import "import pylint" should be placed before "import module2" (wrong-import-order)
```

### Command used

```shell
pylint ${HOME}/root_dir/repo_dir/src/module1 ${HOME}/root_dir/repo_dir/src/module2
```

### Pylint output

<table>
  <tr>
    <td><b>Working directory</b></td>
    <td>root_dir</td>
    <td>repo_dir</td>
  </tr>
  <tr>
    <td><b>Output</b></td>
    <td>
<pre><code>--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)</code></pre>
    </td>
    <td>
<pre><code>************* Module module1.function1
src/module1/function1.py:6:0: C0411: third party import "import pylint" should be placed before "import module2" (wrong-import-order)

-------------------------------------------------------------------
Your code has been rated at 8.89/10 (previous run: 10.00/10, -1.11)</code></pre>
    </td>
  </tr>
</table>

### Expected behavior

`pylint` output should be the same regardless of where it was run from so long as settings are the same

### Pylint version

```shell
pylint 2.17.4
astroid 2.15.5
Python 3.11.3 (main, Jun  7 2023, 20:38:35) [GCC 9.4.0]
```

### OS / Environment

```
➜  ~ cat /etc/os-release     
NAME="Ubuntu"
VERSION="20.04.6 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.6 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
➜  ~ uname -r           
5.15.0-72-generic
➜  ~ pdm --version                                               
PDM, version 2.7.0
```

### Additional dependencies

Compiled with `pdm update --project root_dir/repo_dir`

--- a/doc/data/messages/w/wrong-import-order/details.rst
+++ b/doc/data/messages/w/wrong-import-order/details.rst
@@ -1,0 +1,25 @@
+Pylint uses ``isort`` to classify imports into standard library,
+third-party, first-party, and local import groups.
+
+``isort`` detects first-party imports by looking for packages relative to
+the current working directory. As a result, running pylint from different
+directories can change how the same import is classified.
+
+For example, with the following project layout::
+
+    project/
+        src/
+            my_package/
+                __init__.py
+                module.py
+
+An import of ``my_package`` in ``module.py`` may be treated as first-party
+when pylint is run from ``project/src``, but as third-party when pylint is
+run from ``project``.
+
+If this warning fires, or fails to fire, inconsistently between runs, set
+``known-first-party`` in your pylint configuration to make the
+classification deterministic::
+
+    [tool.pylint.imports]
+    known-first-party = ["my_package"]

--- a/doc/whatsnew/fragments/8801.other
+++ b/doc/whatsnew/fragments/8801.other
@@ -1,0 +1,5 @@
+Document that the ``wrong-import-order`` (C0411) classification of imports as
+third-party vs first-party depends on the current working directory and
+recommend ``known-first-party`` as the deterministic workaround.
+
+Closes #8801

